tests/gcloud-testing-core/src/main/java/com/google/cloud/runtime/jetty/test/AbstractIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtime.jetty.test;
+
+import com.google.cloud.runtime.jetty.util.AppDeployment;
+import com.google.cloud.runtime.jetty.util.HttpUrlUtil;
+
+import org.junit.Assert;
+import org.junit.BeforeClass;
+
+import java.net.URI;
+import java.util.concurrent.TimeUnit;
+
+public class AbstractIntegrationTest {
```

Comment:
I tinkered with that but the problem is the true base class we want is the one that actually implements the test logic and validation and without multiple inheritance...it isn't possible to split these.